src/core/memory.cpp

```diff
@@ -268,6 +269,114 @@ int MemoryManager::QueryProtection(VAddr addr, void** start, void** end, u32* pr
     return ORBIS_OK;
 }
 
+int MemoryManager::Protect(VAddr addr, size_t size, MemoryProt prot) {
+    std::scoped_lock lk{mutex};
+
+    // Find the virtual memory area that contains the specified address range.
+    auto it = FindVMA(addr);
+    if (it == vma_map.end() || !it->second.Contains(addr, size)) {
+        LOG_ERROR(Core, "Address range not mapped");
+        return ORBIS_KERNEL_ERROR_EINVAL;
+    }
+
+    VirtualMemoryArea& vma = it->second;
+    if (vma.type == VMAType::Free) {
+        LOG_ERROR(Core, "Cannot change protection on free memory region");
+        return ORBIS_KERNEL_ERROR_EINVAL;
+    }
+
+    // Validate protection flags
+    MemoryProt valid_flags = MemoryProt::NoAccess | MemoryProt::CpuRead | MemoryProt::CpuReadWrite |
+                             MemoryProt::GpuRead | MemoryProt::GpuWrite | MemoryProt::GpuReadWrite;
+
+    if ((prot & ~valid_flags) != MemoryProt::NoAccess) {
+        LOG_ERROR(Core, "Invalid protection flags, prot: {:#x}, GpuWrite: {:#x}",
+                  static_cast<uint32_t>(prot), static_cast<uint32_t>(MemoryProt::GpuWrite));
```

Comment:
The GpuWrite logging doesn't make sense, it will always log the value of GpuWrite enum? What is the intention here?
```suggestion
                  u32(prot), u32(MemoryProt::GpuWrite));
```

Reply:
Its redirecting to through the api to Gpu Anum